src/UI/SignTransactionsModals/index.tsx

```diff
@@ -41,68 +22,39 @@ export const SignTransactionsModals = ({
   CustomConfirmScreens,
   verifyReceiverScam = true
 }: SignTransactionsPropsType) => {
-  const {
-    callbackRoute,
-    transactions,
-    error,
-    sessionId,
-    onAbort,
-    hasTransactions
-  } = useSignTransactions();
-
-  const { providerType } = useGetAccountProvider();
-  const signTransactionsError = useGetSignTransactionsError();
   const { loginMethod } = useGetLoginInfo();
 
-  const handleClose = () => {
-    onAbort(sessionId);
+  const ConfirmScreens: CustomConfirmScreensType = {
+    Ledger: CustomConfirmScreens?.Ledger ?? SignWithLedgerModal,
+    WalletConnect:
+      CustomConfirmScreens?.WalletConnect ?? SignWithWalletConnectModal,
+    Extension: CustomConfirmScreens?.Extension ?? SignWithExtensionModal,
+    Extra: CustomConfirmScreens?.Extra
   };
 
-  const signError = error || signTransactionsError;
-
-  const signProps: SignPropsType = {
-    handleClose,
-    error: signError,
-    sessionId,
-    transactions: transactions!,
-    providerType,
-    callbackRoute,
-    className,
-    verifyReceiverScam
-  };
-
-  if (signError || hasTransactions) {
-    switch (loginMethod) {
-      case LoginMethodsEnum.ledger:
-        return CustomConfirmScreens?.Ledger ? (
-          <CustomConfirmScreens.Ledger {...signProps} />
-        ) : (
-          <SignWithLedgerModal {...signProps} />
-        );
-
-      case LoginMethodsEnum.walletconnect:
-        return CustomConfirmScreens?.WalletConnect ? (
-          <CustomConfirmScreens.WalletConnect {...signProps} />
-        ) : (
-          <SignWithWalletConnectModal {...signProps} />
-        );
-
-      case LoginMethodsEnum.extension:
-        return CustomConfirmScreens?.Extension ? (
-          <CustomConfirmScreens.Extension {...signProps} />
-        ) : (
-          <SignWithExtensionModal {...signProps} />
-        );
-
-      case LoginMethodsEnum.extra:
-        return CustomConfirmScreens?.Extra ? (
-          <CustomConfirmScreens.Extra {...signProps} />
-        ) : null;
-
-      default:
-        return null;
-    }
+  const renderScreen = useCallback(
+    (Screen?: ScreenType) => {
+      return (
+        <ConfirmationScreen
+          Screen={Screen}
+          verifyReceiverScam={verifyReceiverScam}
+          className={className}
+        />
+      );
+    },
+    [verifyReceiverScam, className]
+  );
+
+  switch (loginMethod) {
+    case LoginMethodsEnum.ledger:
+      return renderScreen(ConfirmScreens.Ledger);
```

Comment:
instead of renderSCreen, wouldn't it be easier to just render the <ConfirmationScreen.../>